tests/src/common/test_case/builder.rs

```diff
@@ -207,17 +207,22 @@ where
             _lpn,
         } = self;
 
-        let profit_addr: Addr = ProfitInstantiator::instantiate(
+        let mut profit_response = ProfitInstantiator::instantiate(
             &mut test_case.app,
             cadence_hours,
             test_case.address_book.treasury().clone(),
             test_case.address_book.oracle().clone(),
             test_case.address_book.time_alarms().clone(),
         );
 
+        profit_response.expect_register_ica(TestCase::DEX_CONNECTION_ID, TestCase::PROFIT_ICA_ID);
+
+        let profit_addr = profit_response.unwrap_response();
+
```

Comment:
For the sake of getting higher cohesion, we want to do it close to the instantiation, ideally right after it. 
The ` ProfitInstantiator::instantiate` would not change the return type as it was before this PR.